prometheus/http.go

```diff
@@ -15,14 +15,169 @@ package prometheus
 
 import (
 	"bufio"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"log"
 	"net"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	contentTypeHeader     = "Content-Type"
+	contentLengthHeader   = "Content-Length"
+	contentEncodingHeader = "Content-Encoding"
+	acceptEncodingHeader  = "Accept-Encoding"
+)
+
+// Handler returns an HTTP handler for the DefaultRegistry. It is
+// already instrumented with InstrumentHandler (using "prometheus" as handler
+// name).
+//
+// Please note the issues described in the doc comment of InstrumentHandler. You
+// might want to consider using UninstrumentedHandler instead.
+//
+// The returned Handler is using the same HandlerOpts as the Handler returned by
+// UninstrumentedHandler. See there for details.
+func Handler() http.Handler {
```

Comment:
I doubt that `Handler` will ever go away. That's just too convenient. A deprecation notice about the handler instrumentation is already in place. I just cannot put a DEPRECATED marker on any of Handler/HandlerFor/UninstrumentedHandler right now, because people _should_ use UnistrumentedHandler right now, but then switch to Handler once it is not instrumented anymore (or instrumented in a saner fashion).

Let's solve that later.

I think the number of top-level identifiers can only be reduced by re-packaging. We can consider that later, see commit description.